elfi/distributions.py

```diff
@@ -32,6 +32,10 @@ def pdf(self, x, *params, **kwargs):
     def logpdf(self, x, *params, **kwargs):
         raise NotImplementedError
 
+    @property
+    def name(self):
+        return self._name or self.__class__.__name__
+
 
```

Comment:
My concern is that if we have distributions that aren't derived from this class and don't have the `name` property. I think it would be better to rather change the `__str__` mehod or what do you think?